.github/workflows/ci.yml

```diff
@@ -5,46 +5,49 @@ on:
   pull_request:
 
 env:
-  CONSUL_VERSION: 1.6.10
-
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+    
 jobs:
-
-  # Enforces the consistency of code formatting using `.editorconfig` and the `dotnet-format` tool.
-  check-format:
+  Consul_AspNetCore:
     if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
-    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, windows-2019, macos-10.15]
+        framework: [netcoreapp3.1]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@v2
```

Comment:
I think we should add a step to install the latest .NET 5.0.x SDK to have consistency between operating systems, otherwise we will be at the mercy of the runners being upgraded at their own pace.

Reply:
Hmm we don't test net5.0 for consul so that is why it is not installed. Added it.

Reply:
Because of `global.json`, we were building it with the .NET 5 SDK, hence my comment.
But it's even better that we test it with .NET 5 now!